### PR TITLE
dd-agent: remove

### PR DESCRIPTION
**Pull request by Sohalt**
###### Description of changes

Superseeds #170423

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/dd-agent/5.nix [hunk 1]
```diff
@@ -1,110 +0,0 @@
...
-  pname = "dd-agent";
-
-  src = fetchFromGitHub {
-    owner  = "datadog";
-    repo   = "dd-agent";
-    rev    = version;
-    sha256 = "1iqxvgpsqibqw3vk79158l2pnb6y4pjhjp2d6724lm5rpz4825lx";
-  };
-
-  patches = [ ./40103-iostat-fix.patch ];
```
Comment (bcdarwin) [inline pkgs/tools/networking/dd-agent/5.nix:50]: Please remove this .patch as it's no longer needed.

**Review by bcdarwin — CHANGES_REQUESTED** [review]

**Comment by bcdarwin** [comment]
It seems like it might be a good idea to add this to the 22.11 release notes as well.

File: pkgs/tools/networking/dd-agent/5.nix (hunk 1 above)
Comment (Sohalt) [reply pkgs/tools/networking/dd-agent/5.nix:50]: thanks!

**Review by bcdarwin — APPROVED** [review]
LGTM.

**Comment by domenkozar** [comment]
Needs a resolved conflict.

**Pull request merged**